app/assets/src/components/views/bulk_download/BulkDownloadModal.jsx

```diff
@@ -71,6 +74,11 @@ class BulkDownloadModal extends React.Component {
         selectedFields: {},
       });
     }
+    if (
+      prevProps.selectedSampleStatuses !== this.props.selectedSampleStatuses
```

Comment:
FYI, this will only compare object identity and not values

Reply:
That's intentional. I want this to be triggered whenever `selectedSampleStatuses` changes.  `prevProps.selectedSampleStatuses` and `this.props.selectedSampleStatuses` will only point to different objects in React when any of the underlying values in the object change, otherwise they will point to the same object. Saves some work doing a deep comparison, since React has already done so.